src/nvim/types.h

```diff
@@ -13,4 +13,5 @@ typedef unsigned char char_u;
 // Can hold one decoded UTF-8 character.
 typedef uint32_t u8char_T;
 
+typedef struct expand expand_T;
```

Comment:
Probably that works also. But I'm not using expand_T in my tests anywhere. So maybe there is a cleaner solution.
If option.c includes ex_cmd_defs.h will that help?